src/main/java/com/eppo/sdk/EppoClient.java

```diff
@@ -85,38 +87,75 @@ private Optional<EppoValue> getAssignmentValue(
             return Optional.empty();
         }
 
-        // Find matched rule
-        Optional<Rule> rule = RuleValidator.findMatchingRule(subjectAttributes, configuration.getRules());
-        if (rule.isEmpty()) {
-            log.info("[Eppo SDK] No assigned variation. The subject attributes did not match any targeting rules");
-            return Optional.empty();
-        }
-
-        // Check if in experiment sample
-        String allocationKey = rule.get().getAllocationKey();
-        Allocation allocation = configuration.getAllocation(allocationKey);
-        if (!this.isInExperimentSample(subjectKey, flagKey, configuration.getSubjectShards(),
-                allocation.getPercentExposure())) {
-            log.info("[Eppo SDK] No assigned variation. The subject is not part of the sample population");
-            return Optional.empty();
+        // Used to assign
+        List<Variation> variations;
+
+        // Used for logging
+        String assignmentModelVersion = "sharding v1";
+        String allocationKey;
+        EppoAttributes assignmentAttributes = null;
+
+        if (configuration.isBandit()) {
+            allocationKey = "bandit";
+
+            // Properties of the bandit model are hardcoded for now
+            String modelName = "random";
+            String modelVersion = "0.1";
+
+            assignmentModelVersion = modelName+" "+modelVersion;
+
+            variations = BanditEvaluator.evaluateBanditVariations(
+              flagKey,
+              modelName,
+              assignmentOptions,
+              subjectKey,
+              subjectAttributes,
+              configuration.getSubjectShards()
+            );
+        } else {
+            // Find matched rule
+            Optional<Rule> rule = RuleValidator.findMatchingRule(subjectAttributes, configuration.getRules());
+            if (rule.isEmpty()) {
+                log.info("[Eppo SDK] No assigned variation. The subject attributes did not match any targeting rules");
+                return Optional.empty();
+            }
+
+            // Check if in experiment sample
+            allocationKey = rule.get().getAllocationKey();
```

Comment:
Note: we may want to enable bandits to use targeting rules. If so, these will be pulled out before the flow control fork for getting variations.